tsl/src/bgw_policy/drop_chunks_api.c

```diff
@@ -213,14 +213,41 @@ drop_chunks_add_policy(PG_FUNCTION_ARGS)
 Datum
 drop_chunks_remove_policy(PG_FUNCTION_ARGS)
 {
-	Oid hypertable_oid = PG_GETARG_OID(0);
+	Oid table_oid = PG_GETARG_OID(0);
 	bool if_exists = PG_GETARG_BOOL(1);
+	Cache *hcache;
+
+	Hypertable *hypertable = ts_hypertable_cache_get_cache_and_entry(table_oid, true, &hcache);
+	if (!hypertable)
+	{
+		char *schema_name = get_namespace_name(get_rel_namespace(table_oid));
+		char *view_name = get_rel_name(table_oid);
+		ContinuousAgg *ca = ts_continuous_agg_find_by_view_name(schema_name, view_name);
+		if (!ca)
+		{
+			if (view_name)
+				ereport(ERROR,
+						(errcode(ERRCODE_UNDEFINED_OBJECT),
+						 errmsg("no hypertable or continuous aggregate by the name \"%s\" exists",
+								view_name)));
+			else
+				ereport(ERROR,
```

Comment:
for an invalid oid schema name and view name will be NULL so the call to ts_continuous_agg_find_by_view_name doesnt make sense so i think this error should be above that call, and maybe include oid in output but i'll leave that up to you

Reply:
Fixed.